packages/dex-core/src/extension/TokenExtension.ts

```diff
@@ -17,41 +14,67 @@ export interface SpendIncreaseProps {
 const sleep = ms => new Promise(done=>setTimeout(done, ms));
 const bn = ethers.BigNumber.from;
 
-export default class TokenSupport {
+export class TokenExtension {
 
-    signer: ethers.Signer;
-    provider: ethers.providers.Provider;
-    address: string | undefined;
-    apiClient: Services.APIClient;
+    apiClient: APIClient;
     chainId: number;
     gnsosisSafe?: string;
+    provider: ethers.providers.Provider;
+    signer?: ethers.Signer;
+    signerAddress: string | undefined;
 
-    constructor(props:ConstructorProps) {
+    constructor(props: APIExtensionProps) {
         this.signer = props.signer;
         this.provider = props.provider;
         this.apiClient = props.apiClient;
         this.gnsosisSafe = props.gnosisSafe;
         this.chainId = 0;
     }
 
-    lookup = async (address:string): Promise<Token> => {
+    protected getSigner() : ethers.Signer {
+        const signer = this.signer;
+        if (! signer) {
+            throw new Error('ethers.Signer is required');
+        }
+        return signer;
+    }
+
+    protected async getSignerAddress() : Promise<string> {
+        const signer = this.getSigner();
+        let signerAddress = this.gnsosisSafe || this.signerAddress;
+
+        if (! this.signerAddress) {
+            signerAddress = await signer.getAddress();
+            this.signerAddress = signerAddress;
+        }
+
+        if (! signerAddress) {
+            throw new Error('Failed to resolve Signer address');
+        }
+
+        this.signerAddress = signerAddress;
+
+        return signerAddress;
+    }
+
+    async lookup(address:string): Promise<Token> {
         let r = await this.verify(address);
         if(!r || r.error) {
             throw new Error("Unsupported token address:" + address);
         }
-        if(!this.address) {
-            this.address = await this.signer.getAddress();
-        }
-        let addr = this.gnsosisSafe || this.address;
+        const owner = await this.getSignerAddress();
+
         return TokenServices.TokenFinder({
             address,
-            owner: addr,
+            owner,
             provider: this.provider
         });
     }
 
-    increaseSpending = async (props:SpendIncreaseProps): Promise<any> => {
-        let network = await this.signer.provider?.getNetwork();
+    async increaseSpending(props:SpendIncreaseProps): Promise<any> {
```

Comment:
I think this should change to "approveSpending". I think my original idea was to increase spending by some amount but it ended up just being an approval. I think we change the name.